txpool/txpool.go

```diff
@@ -25,12 +26,18 @@ const (
 )
 
 var (
-	ErrIntrinsicGas = errors.New("intrinsic gas too low")
+	ErrIntrinsicGas      = errors.New("intrinsic gas too low")
+	ErrNegativeValue     = errors.New("negative value")
+	ErrNonEncryptedTxn   = errors.New("non-encrypted transaction")
+	ErrInvalidSender     = errors.New("invalid sender")
+	ErrNonceTooLow       = errors.New("nonce too low")
+	ErrInsufficientFunds = errors.New("insufficient funds for gas * price + value")
```

Comment:
I'll try to see if I can add the check for `ErrAlreadyKnown` and `ErrReplaceUnderpriced` 👍 

`ErrUnderpriced` is unsupported because we don't have any configurable transaction pool minimums.
`ErrTxPoolOverflow` is unsupported because we don't enforce any txpool limits (whereas geth does)
`ErrGasLimit` is already part of the block limit task 👍 
`ErrOversizedData` is unsupported because we don't enforce transaction sizes in the txpool

I'll try to write individual tests for reach check, just so we know they're being handled properly, even though we've seen them work indirectly 👍

Reply:
I've added tests covering all txpool error codes, apart from `ErrAlreadyKnown` and `ErrReplaceUnderpriced`.
`ErrAlreadyKnown` is an error I believe we can never have due to our current txpool add flow.
`ErrReplaceUnderpriced` would require a refactor for our current txpool add flow, which should be a part of much bigger task (created on YouTrack).

Added unit tests where applicable, and e2e tests for error codes which required finer state management 👍 
2a26ae9

Reply:
GJ
I've added tasks for other errors, to have them in the backlog.